engine/htmlhint/reporter.js

```diff
@@ -0,0 +1,54 @@
+const template = module.exports = {
+    run: function(onFinish) {
+        process.stdin.resume();
+        process.stdin.setEncoding("utf8");
+
+        let lingeringLine = "";
+        let lines = [];
+
+        process.stdin.on("data", (chunk) => {
+            let linesParsed = chunk.split("\n");
+            linesParsed[0] = lingeringLine + linesParsed[0];
+            lingeringLine = linesParsed.pop();
+            linesParsed.forEach((line) => {
+                if (line !== null) {
+                    lines.push(line);
+                }
+            });
+        });
+
+        process.stdin.on("end", () => {
+            onFinish(lines);
+        });
+    },
+};
+
+template.run((lines) => {
+    let results = [];
+
+    lines.forEach((line) => {
+        if (line === "") {
+            return;
+        }
+    
+        let data = JSON.parse(line);
+        results = data.map((current) => {
+            return {
+                path: current.file,
+                messages: current.messages.map((problem) => {
+                    return {
+                        message: problem.message,
+                        severity: problem.type,
+                        source: problem.evidence.trim(),
+                        line: problem.line - 1,
+                        lineEnd: problem.line - 1,
+                        column: problem.col - 1,
+                        ruleId: problem.rule.id,
+                    };
+                }),
+            };
+        });
+    });
+
+    console.log(JSON.stringify(results));
```

Comment:
I also suggest to return an object from this method and perform conversion to JSON+console.write at least in the upper level.

Reply:
done